src/Native/Runtime/coreclr/gcinfodecoder.cpp

```diff
@@ -1476,23 +1476,27 @@ OBJECTREF* GcInfoDecoder::GetRegisterSlot(
     _ASSERTE(regNum >= 0 && regNum <= 14);
     _ASSERTE(regNum != 13);  // sp
 
+#ifdef FEATURE_REDHAWK
+    PTR_UIntNative *ppReg;
+#else
     DWORD **ppReg;
+#endif
 
     if(regNum <= 3)
     {
-        ppReg = &pRD->volatileCurrContextPointers.R0;
+        ppReg = &pRD->pR0;
         return (OBJECTREF*)*(ppReg + regNum);
     }
```

Comment:
Could you please put the whole block with changes under one larger `#ifdef FEATURE_REDHAWK` so that the file can be easily shared with CoreCLR if necessary?